src/main/kotlin/com/terraformation/backend/search/table/SpeciesGrowthFormsTable.kt

```diff
@@ -0,0 +1,37 @@
+package com.terraformation.backend.search.table
+
+import com.terraformation.backend.db.default_schema.SpeciesId
+import com.terraformation.backend.db.default_schema.tables.references.SPECIES
+import com.terraformation.backend.db.default_schema.tables.references.SPECIES_GROWTH_FORMS
+import com.terraformation.backend.search.SearchTable
+import com.terraformation.backend.search.SublistField
+import com.terraformation.backend.search.field.SearchField
+import org.jooq.Record
+import org.jooq.SelectJoinStep
+import org.jooq.TableField
+
+class SpeciesGrowthFormsTable(private val tables: SearchTables) : SearchTable() {
+  override val primaryKey: TableField<out Record, out Any?>
+    get() = SPECIES_GROWTH_FORMS.SPECIES_ID
+
+  override val sublists: List<SublistField> by lazy {
+    with(tables) {
+      listOf(
+          species.asSingleValueSublist("species", SPECIES_GROWTH_FORMS.SPECIES_ID.eq(SPECIES.ID)),
+      )
+    }
+  }
+
+  override val fields: List<SearchField> =
+      listOf(
+          idWrapperField("species_id", SPECIES_GROWTH_FORMS.SPECIES_ID) { SpeciesId(it) },
```

Comment:
For some reason this line `SPECIES_GROWTH_FORMS.SPECIES_ID` is saying it needs to be `Int` instead of `SpeciesId`. I was hoping [this line](https://github.com/terraware/terraware-server/blob/main/jooq/src/main/kotlin/com/terraformation/backend/jooq/Config.kt#L232) would figure out the correct type for the field since it is `species_growth_forms.species_id`. @sgrimm  Am I missing something?

Reply:
Changing the column type should fix the problem with this being `Int`, but another problem here is the field name. Periods and underscores are reserved characters in the search API. `species_id` means, "the `id` field from the `species` sublist."

You have two options here.

* Get rid of this field entirely since you already have `species` as a sublist, and then use `species_id` which will join with the species table and give you the ID. That's the approach we use for foreign keys in most of the search tables.
* Keep an explicit ID field but call it `speciesId` so it's not parsed as a sublist field reference. This is slightly more performant if you're doing a lot of queries where you only care about species IDs but you don't care about any other species fields. If you also want, say, species names, then having an explicit `speciesId` field won't make things any faster since the query will have to join with the species table regardless.

My hunch is that you'll mostly be accessing this table as a sublist of the species table, rather than starting from here and looking up species information. In that case, you definitely won't need an explicit ID field since nothing would access it anyway.

Reply:
You'll also want to add this as a sublist on the species search table so you can query the growth forms for a species.

Reply:
I realized there was no validation of field names to make sure they don't run into ambiguity problems, so I added some test coverage in #1882 .

Reply:
Thanks for the info, I went with the first option.

Reply:
> You'll also want to add this as a sublist on the species search table so you can query the growth forms for a species.

Is [this change](https://github.com/terraware/terraware-server/pull/1880/files#diff-6a2f0959bf379022d4c70bec7f494557b26cf0f1978f6c5266f942cfaa2fd417R34-R35) sufficient for that, or did I miss something else?

Reply:
Oh, oops, somehow I managed to not see that. Yes, that's totally sufficient.